### DISCUSSION TITLE
Why Can't I stackalloc delegates?

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/runtime#71618] [Q&A] [answered]
```
            Span< λ > k = new λ[5]; //works
            Span< λ > k = stackalloc λ[5]; //doesn't compile
```

**Comment by heathbm** [comment]
The error message my IDE provides is: "Cannot declare a pointer to a managed type 'Delegate'"

**Comment by huoyaoyuan** [comment, marked as answer]
You can't `stackalloc` for any reference type either.
The direct reason is that the result of `stackalloc` IL instruction is a pointer. The span is synthesized by C# compiler.
There is already a proposal to enable `stackalloc` for reference type: #11192

**Comment by BreyerW** [comment]
Also you can get function pointer directly from static method if you care about perf. Not array of func pointers tho afaik
